### PR TITLE
chore(homework-5): code refactoring

### PR DESCRIPTION
**Pull request by AntoniaSzecsi**

### CODE REVIEW

File: sensors/Sensor.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        virtual ~Sensor();
+        const std::string &getName() const;
+        int getValue() const;
+        MeasureUnit getUnit() const;
+
+        virtual void generateRandomValue() = 0;
+
+    protected:
+        std::string name;
+        int *value;
```
Comment (CristianIonutIonescu) [inline sensors/Sensor.h]: why are you using an int here when all your sensors use doubles? you should change this to integer unless you have other plans for it.

File: utils/StatusPrinter.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#define NAGARROREMOTELEARNING_STATUSPRINTER_H
+
+#include <vector>
+#include "../devices/Device.h"
+#include "../sensors/Sensor.h"
+
+namespace SmartHome {
+    class StatusPrinter {
+    public:
+        static void printDeviceStatus(std::vector<Device*> &devices);
```
Comment (CristianIonutIonescu) [inline utils/StatusPrinter.h]: Just by looking at the name of the method the parameter should be const ref instead of ref

File: user/User.cc [hunk 3]
```diff
@@ -1,135 +1,83 @@
...
-    }
-
-    User::~User() {
-        for (Device *device: devices) {
-            delete device;
-        }
+    User::User(int numRooms) : userHouse(numRooms) {
+        devices.push_back(new Light());
+        devices.push_back(new Heater());
+        devices.push_back(new AirConditioner());
```
Comment (CristianIonutIonescu) [inline user/User.cc:11]: you should have left the previous implementation here as now you have memory leaks due to not deleting the devices.

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]
looks great, needs a bit more polish but overall good! Good job

File: user/User.cc (hunk 3 above)
Comment (AntoniaSzecsi) [reply user/User.cc:11]: Will add a destructor to the User class

File: user/User.h [hunk 4]
```diff
@@ -11,6 +11,7 @@ namespace SmartHome {
     class User {
     public:
         explicit User(int numRooms);
+        ~User();
```
Comment (CristianIonutIonescu) [inline user/User.h:14]: adding just the destructor your missed the rule of 5 :(

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**